scheduling-manager/src/main/java/com/amazonaws/blox/scheduling/scheduler/engine/daemon/ReplaceAfterTerminateScheduler.java

```diff
@@ -15,18 +15,29 @@
 package com.amazonaws.blox.scheduling.scheduler.engine.daemon;
 
 import com.amazonaws.blox.scheduling.scheduler.engine.SchedulingAction;
+import java.util.ArrayList;
 import java.util.List;
 import java.util.stream.Collectors;
 
 public class ReplaceAfterTerminateScheduler extends DaemonScheduler {
   public static final String ID = "ReplaceAfterTerminate";
 
   public List<SchedulingAction> schedule(DaemonEnvironment env, ClusterSummary summary) {
-    return summary
+
+    List<SchedulingAction> stopTaskActions = new ArrayList<>();
+    summary
         .getInstances()
         .stream()
-        .filter(i -> env.hasMatchingTask(summary.tasksForInstance(i)))
-        .map(i -> env.startTaskFor(i))
-        .collect(Collectors.toList());
+        .filter(i -> env.hasUnmatchingTask(summary.tasksForInstance(i)))
+        .forEach(i -> stopTaskActions.addAll(env.stopTaskFor(summary.tasksForInstance(i))));
```

Comment:
This can be written with a `flatMap` instead of imperatively modifying `stopTaskActions` using `forEach`.